src/db/spanner/manager/deadpool.rs

```diff
@@ -52,24 +55,47 @@ impl SpannerSessionManager {
             env,
             metrics: metrics.clone(),
             test_transactions,
+            max_lifespan: settings.database_pool_connection_lifespan,
+            max_idle: settings.database_pool_connection_max_idle,
         })
     }
 }
 
 #[async_trait]
 impl Manager<SpannerSession, DbError> for SpannerSessionManager {
     async fn create(&self) -> Result<SpannerSession, DbError> {
-        create_spanner_session(
+        let session = create_spanner_session(
             Arc::clone(&self.env),
             self.metrics.clone(),
             &self.database_name,
             self.test_transactions,
         )
-        .await
+        .await?;
+        // check how long that this has been idle...
+        if let Some(max_idle) = self.max_idle {
+            if let Some(idle) = session
+                .session
+                .approximate_last_use_time
+                .clone()
+                .into_option()
+            {
+                // get current UTC seconds
+                let now = SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap_or_default()
+                    .as_secs();
+                let idle = std::cmp::max(0, now as i64 - idle.seconds);
```

Comment:
Is there a reason for i64 here vs u64 in the max_lifetime check code?

Keeping everything as a higher level `Duration` type might be a bit nicer and avoid some of this casting -- except the grpc `Timeout` uses signed values whereas `Duration` uses unsigned, sigh.

Reply:
yeah, pretty much that's the reason.